addons/account/report/account_bank_reconciliation_report.py

```diff
@@ -0,0 +1,204 @@
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api, _
+
+
+class account_bank_reconciliation(models.AbstractModel):
+    _name = 'account.bank.reconciliation'
+    _description = 'Bank reconciliation report'
+
+    line_number = 0
+
+    @api.model
+    def get_lines(self, context_id, line_id=None):
+        return self.with_context(
+            date_from=context_id.date_from,
+            journal_id=context_id.journal_id,
+            context_id=context_id,
+        )._lines()
+
+    def add_title_line(self, title, amount):
+        self.line_number += 1
+        return {
+            'id': self.line_number,
+            'type': 'line',
+            'name': title,
+            'footnotes': self._get_footnotes('line', self.line_number),
+            'columns': [self.env.context['date_from'], '', amount],
+            'level': 0,
+        }
+
+    def add_subtitle_line(self, title):
+        self.line_number += 1
+        return {
+            'id': self.line_number,
+            'type': 'line',
+            'name': title,
+            'footnotes': self._get_footnotes('line', self.line_number),
+            'columns': [self.env.context['date_from'], '', ''],
+            'level': 1,
+        }
+
+    def add_total_line(self, amount):
+        self.line_number += 1
+        return {
+            'id': self.line_number,
+            'type': 'line',
+            'name': '',
+            'footnotes': self._get_footnotes('line', self.line_number),
+            'columns': [self.env.context['date_from'], "", _("Total : ") + str(amount)],
+            'level': 2,
+        }
+
+    def add_bank_statement_line(self, line, amount):
+        self.line_number += 1
+        return {
+            'id': self.line_number,
+            'statement_id': line.statement_id.id,
+            'type': 'bank_statement_id',
+            'name': line.name,
+            'footnotes': self._get_footnotes('bank_statement_id', self.line_number),
+            'columns': [line.date, line.ref, amount],
+            'level': 3,
+        }
+
+    @api.model
+    def _lines(self):
+        lines = []
+        #Start amount
+        account_ids = list(set([self.env.context['journal_id'].default_debit_account_id.id, self.env.context['journal_id'].default_credit_account_id.id]))
+        lines_already_accounted = self.env['account.move.line'].search(['|', ('move_id.journal_id.type', '=', 'general'),
```

Comment:
no, query_get() should only used when there is a need of using sql query directly... Here, it wouldn't add any value